packages/react-dom-bindings/src/client/CSSPropertyOperations.js

```diff
@@ -167,15 +187,38 @@ function expandShorthandMap(styles) {
  *   becomes .style.fontVariant = ''
  */
 export function validateShorthandPropertyCollisionInDev(
-  styleUpdates,
+  prevStyles,
   nextStyles,
 ) {
   if (__DEV__) {
     if (!nextStyles) {
       return;
     }
 
-    const expandedUpdates = expandShorthandMap(styleUpdates);
+    // Compute the diff as it would happen elsewhere.
+    const expandedUpdates = {};
+    if (prevStyles) {
+      for (const key in prevStyles) {
+        if (prevStyles.hasOwnProperty(key) && !nextStyles.hasOwnProperty(key)) {
+          const longhands = shorthandToLonghand[key] || [key];
+          for (let i = 0; i < longhands.length; i++) {
+            expandedUpdates[longhands[i]] = key;
+          }
+        }
+      }
+    }
+    for (const key in nextStyles) {
+      if (
+        nextStyles.hasOwnProperty(key) &&
+        (!prevStyles || prevStyles[key] !== nextStyles[key])
+      ) {
+        const longhands = shorthandToLonghand[key] || [key];
+        for (let i = 0; i < longhands.length; i++) {
+          expandedUpdates[longhands[i]] = key;
+        }
+      }
+    }
```

Comment:
@sophiebits I tried to be more clever but this is as far as I got.

Reply:
Are you setting all the style properties on every render now?

Reply:
It diffs the React style object before applying it to the DOM. So it’s the same, just that it’s done later and without an intermediate object that contains nulled out fields.

Reply:
Hm. You’re right I probably shouldn’t do that and would need to add diffing to setStyles.

Unclear if that is worth it though.

Reply:
Fixed.